### ISSUE TITLE
[BUG] IllegalArgumentException when use KafkaConsumer#offsetsForTimes

### ISSUE THREAD

**Issue by aloyszhang** [streamnative/kop#135]
**Describe the bug**
When tries KOP with kafka 2.1.0,2.2.0 and 2.3.0, if  `KafkaConsumer#offsetsForTimes` invoked, we'll have the exception follows
```
Exception in thread "main" java.lang.IllegalArgumentException: Invalid negative timestamp
        at org.apache.kafka.clients.consumer.OffsetAndTimestamp.<init>(OffsetAndTimestamp.java:39)
        at org.apache.kafka.clients.consumer.internals.Fetcher.offsetsForTimes(Fetcher.java:500)
        at org.apache.kafka.clients.consumer.KafkaConsumer.offsetsForTimes(KafkaConsumer.java:2009)
        at org.apache.kafka.clients.consumer.KafkaConsumer.offsetsForTimes(KafkaConsumer.java:1973)
        at ConsumerKafka.lambda$compatibleTest$0(ConsumerKafka.java:94)
        at ConsumerKafka.tryFeature(ConsumerKafka.java:179)
        at ConsumerKafka.compatibleTest(ConsumerKafka.java:93)
        at ConsumerKafka.main(ConsumerKafka.java:62)
```

**To Reproduce**
1. run pulsar on KOP 
2. start a kafka consumer under version 2.1.0,2.2.0 and 2.3.0
3. invoke `KafkaConsumer#offsetsForTimes`

**Expected behavior**
`KafkaConsumer#offsetsForTimes` should work well

**Additional context**
When I do troubleshooting, I found KOP always return offset informations with timestamp -1. 
```
 partitionData.complete(new ListOffsetResponse.PartitionData(
                     Errors.NONE,
                     RecordBatch.NO_TIMESTAMP, \\ -1
                     MessageIdUtils.getOffset(finalPosition.getLedgerId(), finalPosition.getEntryId())));
```
But in Kafka java client of 2.1.0,2.2.0,2.3.0, there is a parameter check on the timestamp which must be bigger than 0.
```
for (Map.Entry<TopicPartition, ListOffsetData> entry : fetchedOffsets.entrySet()) {
                // 'entry.getValue().timestamp' will not be null since we are guaranteed
                // to work with a v1 (or later) ListOffset request
                ListOffsetData offsetData = entry.getValue();
                offsetsByTimes.put(entry.getKey(), new OffsetAndTimestamp(offsetData.offset, offsetData.timestamp,
                        offsetData.leaderEpoch));
            }

...

public OffsetAndTimestamp(long offset, long timestamp, Optional<Integer> leaderEpoch) {
        if (offset < 0)
            throw new IllegalArgumentException("Invalid negative offset");

        if (timestamp < 0)
            throw new IllegalArgumentException("Invalid negative timestamp");

        this.offset = offset;
        this.timestamp = timestamp;
        this.leaderEpoch = leaderEpoch;
    }
```
This is the cause of the exception.

**Comment by jiazhai** [comment]
Thanks @aloyszhang for the help. Would you like to provide a fix for this?

**Comment by aloyszhang** [comment]
@jiazhai I'll  try to fix it

**Comment by sijie** [comment]
This has been fixed by #137
